substrate/frame/nfts/src/features/transfer.rs

```diff
@@ -116,49 +119,53 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	///
 	/// - `origin`: The account requesting the transfer.
 	/// - `collection`: The ID of the collection to transfer ownership.
-	/// - `owner`: The new account that will become the owner of the collection.
+	/// - `new_owner`: The account that will become the new owner of the collection.
 	///
 	/// This function transfers the ownership of a collection to the specified account.
 	/// It performs checks to ensure that the `origin` is the current owner and that the
 	/// new owner is an acceptable account based on the collection's acceptance settings.
 	pub(crate) fn do_transfer_ownership(
 		origin: T::AccountId,
 		collection: T::CollectionId,
-		owner: T::AccountId,
+		new_owner: T::AccountId,
 	) -> DispatchResult {
 		// Check if the new owner is acceptable based on the collection's acceptance settings.
-		let acceptable_collection = OwnershipAcceptance::<T, I>::get(&owner);
+		let acceptable_collection = OwnershipAcceptance::<T, I>::get(&new_owner);
 		ensure!(acceptable_collection.as_ref() == Some(&collection), Error::<T, I>::Unaccepted);
 
 		// Try to retrieve and mutate the collection details.
 		Collection::<T, I>::try_mutate(collection, |maybe_details| {
 			let details = maybe_details.as_mut().ok_or(Error::<T, I>::UnknownCollection)?;
 			// Check if the `origin` is the current owner of the collection.
 			ensure!(origin == details.owner, Error::<T, I>::NoPermission);
-			if details.owner == owner {
+			if details.owner == new_owner {
 				return Ok(())
 			}
 
 			// Move the deposit to the new owner.
-			T::Currency::repatriate_reserved(
+			T::Currency::transfer_on_hold(
+				&HoldReason::CollectionOwnerAggregatedDeposit.into(),
 				&details.owner,
-				&owner,
+				&new_owner,
 				details.owner_deposit,
-				Reserved,
+				BestEffort,
```

Comment:
Shoud this be Precise? I'd think if we couldn't transfer all the deposit we'd prefer to fail the transaction?